### PR TITLE
Added Queue implementations.

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

File: source/collections/Queue.ooc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	//new: static func -> Queue<T> { VectorQueue<T> new() }
+}
+
+VectorQueue: class <T> extends Queue<T> {
+	_backend: T*
+	_capacity := 0
+	_head := 0
+	_tail := 0
+	capacity ::= this _capacity
+	count ::= this _count
```
Comment (davidhesselbom) [inline source/collections/Queue.ooc]: This shouldn't be needed here.

File: source/collections/SynchronizedQueue.ooc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	enqueue: override func (item: T) {
+		super(item)
+		this _populated signal()
+	}
+	wait: func -> T {
+		result: T
+		this _mutex lock()
+		while (this empty) {
+			this _populated wait(this _mutex)
+		}
```
Comment (davidhesselbom) [inline source/collections/SynchronizedQueue.ooc]: No need for these brackets.

File: test/collections/BlockedQueueTest.ooc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		func2 := func {
+			result: Int
+			success := queue peek(result&)
+			expect(success)
+			expect(result == 123)
+		}
+		func3 := func {
+			result: Int
+			success := queue dequeue(result&)
+			if(success)
```
Comment (davidhesselbom) [inline test/collections/BlockedQueueTest.ooc]: space after `if`

File: test/collections/BlockedQueueTest.ooc [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			/* Peek values asynchronously in empty Queue */
+			for (i in 0..20)
+				pool add(func4)
+
+			pool waitAll()
+
+			expect(queue empty)
+			expect(queue count, is equal to(0))
+
+			for(i in 0..4) {
```
Comment (davidhesselbom) [inline test/collections/BlockedQueueTest.ooc]: space after `for`

File: test/collections/BlockedQueueTest.ooc [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+				pool add(func5)
+				Time sleepMilli(10)
+			}
+			for(i in 0..4) {
```
Comment (davidhesselbom) [inline test/collections/BlockedQueueTest.ooc]: space after `for`

File: test/collections/SynchronizedQueueTest.ooc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+		func2 := func {
+			result: Int
+			success := queue peek(result&)
+			expect(success)
+			expect(result == 123)
+		}
+		func3 := func {
+			result: Int
+			success := queue dequeue(result&)
+			if(success)
```
Comment (davidhesselbom) [inline test/collections/SynchronizedQueueTest.ooc]: space after `if`

**Comment by emilwestergren** [comment]
Updated.

**Pull request merged**